app/keepers/keepers.go

```diff
@@ -76,5 +132,455 @@ type AppKeepers struct {
+		appCodec,
+		appKeepers.keys[qoraclemoduletypes.StoreKey],
+		appKeepers.memKeys[qoraclemoduletypes.MemStoreKey],
+		appKeepers.tkeys[qoraclemoduletypes.TStoreKey],
+		appKeepers.GetSubspace(qoraclemoduletypes.ModuleName),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
+	appKeepers.QOsmosisKeeper = qosmokeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[qosmotypes.StoreKey],
+		appKeepers.GetSubspace(qosmotypes.SubModuleName),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		appKeepers.IBCKeeper.ClientKeeper,
+		appKeepers.IBCKeeper.ChannelKeeper,
+		appKeepers.IBCKeeper.ChannelKeeper,
+		&appKeepers.IBCKeeper.PortKeeper,
+		appKeepers.ScopedQOsmosisKeeper,
+		appKeepers.QOracleKeeper,
+	)
+	//qosmoIBCModule := qosmo.NewIBCModule(app.QOsmosisKeeper)
+
+	appKeepers.QOracleKeeper.RegisterPoolOracle(appKeepers.QOsmosisKeeper)
+	appKeepers.QOracleKeeper.Seal()
+	//qoracleModule := qoraclemodule.NewAppModule(appCodec, appKeepers.QOracleKeeper, appKeepers.QOsmosisKeeper)
+
+	appKeepers.QTransferKeeper = qtransferkeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[qtransfertypes.ModuleName],
+		appKeepers.GetSubspace(qtransfertypes.ModuleName),
+		appKeepers.AccountKeeper,
+	)
+	//qtranserModule := qtransfer.NewAppModule(appKeepers.QTransferKeeper)
+
+	appKeepers.QVestingKeeper = *qvestingmodulekeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[qvestingmoduletypes.StoreKey],
+		appKeepers.keys[qvestingmoduletypes.MemStoreKey],
+		appKeepers.GetSubspace(qvestingmoduletypes.ModuleName),
+		appKeepers.AccountKeeper,
+		appKeepers.BankKeeper,
+	)
+
+	//qvestingModule :=
+
+	// Authz
+	appKeepers.AuthzKeeper = authzkeeper.NewKeeper(
+		appKeepers.keys[authzkeeper.StoreKey],
+		appCodec,
+		bApp.MsgServiceRouter(),
+		appKeepers.AccountKeeper,
+	)
+
+	// Set epoch hooks
+	appKeepers.EpochsKeeper.SetHooks(
+		epochsmoduletypes.NewMultiEpochHooks(
+			appKeepers.QOsmosisKeeper.EpochHooks(),
+		),
+	)
+
+	/// Token factory Module
+	appKeepers.TfKeeper = tfkeeper.NewKeeper(appKeepers.keys[tftypes.StoreKey],
+		appKeepers.GetSubspace(tftypes.ModuleName),
+		appKeepers.AccountKeeper,
+		appKeepers.BankKeeper,
+		appKeepers.DistrKeeper,
+	)
+	//tfModule := tfmodule.NewAppModule(appKeepers.TfKeeper,
+	//	appKeepers.AccountKeeper,
+	//	appKeepers.BankKeeper,
+	//)
+
+	callback := owasm.NewCallbackPlugin(appKeepers.WasmKeeper, appKeepers.QTransferKeeper.GetQTransferAcc())
+
+	// AUDIT CHECK IS THIS TYPE ASSERTION FOR TYPE CASTING INTERFACE TO STRUCT SAFE?
+	tmpBankBaseKeeper := appKeepers.BankKeeper.(bankkeeper.BaseKeeper)
+
+	wasmOpts = append(owasm.RegisterCustomPlugins(appKeepers.QOracleKeeper, &tmpBankBaseKeeper, callback), wasmOpts...)
+	wasmOpts = append(tfbindings.RegisterCustomPlugins(&tmpBankBaseKeeper, &appKeepers.TfKeeper), wasmOpts...)
+
+	// The last arguments can contain custom message handlers, and custom query handlers,
+	// if we want to allow any custom callbacks
+	supportedFeatures := "iterator,staking,stargate"
```

Comment:
@arhamchordia 
cosmwasm_1_1,cosmwasm_1_2,cosmwasm_1_4"  are missing in the supported feature.

Reply:
Added